src/apm-cli.coffee

```diff
@@ -100,34 +100,49 @@ printVersions = (args, callback) ->
 
   getPythonVersion (pythonVersion) ->
     git.getGitVersion (gitVersion) ->
-      if args.json
-        versions =
-          apm: apmVersion
-          npm: npmVersion
-          node: nodeVersion
-          python: pythonVersion
-          git: gitVersion
-          nodeArch: process.arch
-        if config.isWin32()
-          versions.visualStudio = config.getInstalledVisualStudioFlag()
-        console.log JSON.stringify(versions)
-      else
-        pythonVersion ?= ''
-        gitVersion ?= ''
-        versions =  """
-          #{'apm'.red}  #{apmVersion.red}
-          #{'npm'.green}  #{npmVersion.green}
-          #{'node'.blue} #{nodeVersion.blue} #{process.arch.blue}
-          #{'python'.yellow} #{pythonVersion.yellow}
-          #{'git'.magenta} #{gitVersion.magenta}
-        """
-
-        if config.isWin32()
-          visualStudioVersion = config.getInstalledVisualStudioFlag() ? ''
-          versions += "\n#{'visual studio'.cyan} #{visualStudioVersion.cyan}"
-
-        console.log versions
-      callback()
+      getAtomVersion (atomVersion) ->
+        if args.json
+          versions =
+            apm: apmVersion
+            npm: npmVersion
+            node: nodeVersion
+            atomVersion: atomVersion
+            python: pythonVersion
+            git: gitVersion
+            nodeArch: process.arch
+          if config.isWin32()
+            versions.visualStudio = config.getInstalledVisualStudioFlag()
+          console.log JSON.stringify(versions)
+        else
+          pythonVersion ?= ''
+          gitVersion ?= ''
+          atomVersion ?= ''
+          versions =  """
+            #{'apm'.red}  #{apmVersion.red}
+            #{'npm'.green}  #{npmVersion.green}
+            #{'node'.blue} #{nodeVersion.blue} #{process.arch.blue}
+            #{'atom'.cyan} #{atomVersion.cyan}
+            #{'python'.yellow} #{pythonVersion.yellow}
+            #{'git'.magenta} #{gitVersion.magenta}
+          """
+
+          if config.isWin32()
+            visualStudioVersion = config.getInstalledVisualStudioFlag() ? ''
+            versions += "\n#{'visual studio'.cyan} #{visualStudioVersion.cyan}"
+
+          console.log versions
+        callback()
+
+getAtomVersion = (callback) ->
+  config.getResourcePath (resourcePath) ->
+    # workaround if app.asar does not exist. see: https://github.com/atom/atom/issues/6604
```

Comment:
is 6604 still a problem?  There was only one report of it.

Reply:
I ran into it while testing. For whatever reason `app.asar` does not exist on my machine.

```shell
~ ❯❯❯ sw_vers
ProductName:	Mac OS X
ProductVersion:	10.11.6
BuildVersion:	15G1611
```
```shell
~/D/p/apm ❯❯❯ ls /Applications/Atom.app/Contents/Resources/                                                                                                                                    atom-version
LICENSE.md    bg.lproj      de.lproj      en_GB.lproj   fi.lproj      he.lproj      it.lproj      lv.lproj      nl.lproj      ru.lproj      sw.lproj      uk.lproj
am.lproj      bn.lproj      el.lproj      es.lproj      fil.lproj     hi.lproj      ja.lproj      ml.lproj      pl.lproj      sk.lproj      ta.lproj      vi.lproj
app           ca.lproj      electron.asar es_419.lproj  file.icns     hr.lproj      kn.lproj      mr.lproj      pt_BR.lproj   sl.lproj      te.lproj      zh_CN.lproj
ar.lproj      cs.lproj      electron.icns et.lproj      fr.lproj      hu.lproj      ko.lproj      ms.lproj      pt_PT.lproj   sr.lproj      th.lproj      zh_TW.lproj
atom.icns     da.lproj      en.lproj      fa.lproj      gu.lproj      id.lproj      lt.lproj      nb.lproj      ro.lproj      sv.lproj      tr.lproj
```

```shell
~/D/p/apm ❯❯❯ apm --version                                                                                                                                                                    atom-version
apm  1.18.4
npm  3.10.10
node 6.9.5 x64
python 2.7.10
git 2.14.2
```

```shell
~/D/p/apm ❯❯❯ atom --version                                                                                                                                                                   atom-version
Atom    : 1.20.1
Electron: 1.6.9
Chrome  : 56.0.2924.87
Node    : 7.4.0
```